console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,26 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// csp is a list of Content Security Policy directives for the plugin.
+	CSP []ConsolePluginCSP `json:"csp"`
+}
+
+// DirectiveType is an enumeration of supported CSP directives
+type DirectiveType string
+
+const (
+	DefaultSrc DirectiveType = "default-src"
+	ScriptSrc  DirectiveType = "script-src"
+	ImgSrc     DirectiveType = "img-src"
+	StyleSrc   DirectiveType = "style-src"
+	FontSrc    DirectiveType = "font-src"
+	BaseUri    DirectiveType = "base-uri"
+)
+
+// ConsolePluginCSP
+type ConsolePluginCSP struct {
+	Directive DirectiveType `json:"directive"`
+	Sources   []string      `json:"source"`
```

Comment:
I would make this something generic like `values` since we might have directives in the future that aren't sources.

Reply:
Need to make clear that this adds to the existing console directive values rather than replacing them.